engine/runtime/src/main/java/org/enso/interpreter/epb/node/ForeignEvalNode.java

```diff
@@ -100,7 +97,7 @@ private void parseJs(ContextReference<EpbContext> ctxRef) {
               + "\n};poly_enso_eval";
       Source source = Source.newBuilder(code.getLanguage().getTruffleId(), wrappedSrc, "").build();
 
-      CallTarget ct = ctxRef.get().getEnv().parsePublic(source);
+      CallTarget ct = context.getEnv().parsePublic(source);
```

Comment:
This is wrong (also, not documented and TBH what you've done looks like a better solution even though it doesn't work at all...). There's a reason why it was `ctxRef.get()` and not `context` before. The reason is, after `inner.enter(this)`, `ctxRef.get()` will return a different context instance. The truffle context is the current state of the "world", including all languages. With `TruffleContext.enter`, we're hot-swapping into a different, isolated truffle instance, where all languages have separate states (i.e. their contexts). We do this so that JS (and other single threaded langs) have a safe, single-threaded context to run their code.

Reply:
Just to be clear – if you run `EpbContext.get(this)` again, it restores the correct semantics.